scio-core/src/main/scala/com/spotify/scio/coders/instances/ZstdCoder.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2024 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.coders.instances
+
+import com.spotify.scio.coders.Coder
+import org.apache.beam.sdk.coders.{ZstdCoder => BZstdCoder}
+
+import scala.reflect.ClassTag
+
+object ZstdCoder {
+  def apply[T: Coder: ClassTag](dict: Array[Byte]): Coder[T] =
+    Coder.transform(Coder[T])(tCoder => Coder.beam(BZstdCoder.of(tCoder, dict)))
+
+  def kv[K: Coder, V: Coder](
+    keyDict: Array[Byte] = null,
+    valueDict: Array[Byte] = null
```

Comment:
I'm not a fan of null defaults.
Also, we used `kv` for `beam.KV` in many placed, not tuples. This signature is a bit confusing.

Do we actually need this API since this is a regular `Tuple2Coder` with implicit (un)compressed key and value coders values ?

Reply:
This is for user ergonomics if setting the coder manually. 
```
sc.parallelize[(String, String)](
  List(...)
)(ZstdCoder.kv(valueDict = dictBytes))
```

Not providing this means users need to manually lift their dict into a Beam coder into a Scio coder. And using optionals here makes the API unpleasant IMO even though I in general agree with you.

Is `ZstdCoder.tuple2` less annoying/confusing?

Reply:
I'm lightly in favor of `tuple2` (unfortunately) -- I think `kv` is already semantically "taken" by Beam `KV` class.